src/main/java/org/takes/facets/auth/codecs/CcAES.java

```diff
@@ -44,77 +43,50 @@
  * AES codec which supports 128 bits key.
  *
  * <p>The class is immutable and thread-safe.
- *
  * @author Jason Wong (super132j@yahoo.com)
  * @version $Id$
  * @since 0.13.8
  */
-@EqualsAndHashCode(of = { "origin", "secret", "enc", "dec" })
+@EqualsAndHashCode(of = {"origin", "key"})
 public final class CcAES implements Codec {
-
     /**
-     * The cipher for encryption.
+     * The block size constant.
      */
-    private final transient Cipher enc;
+    private static final int BLOCK = 16;
 
     /**
-     * The cipher for decryption.
+     * Original codec.
      */
-    private final transient Cipher dec;
+    private final transient Codec origin;
```

Comment:
@mkordas 
- I thought(from my quick and limited understanding of the classes, Back and BkWrap), this is a right name, It give the idea of getting the root codec or the previous one.
 - the variable name was present in the existing code.

But if you think, simple codec is the right choice, I have no issue changing them. I understand, teamed like to make classes more simple :) so I am just double checking :)